scripts/pre_commit_linter.py

```diff
@@ -80,34 +84,67 @@ def _is_javascript_file(filename):
     return filename.endswith('.js')
 
 
-def _lint_js_files(node_path, jscs_path, autofix, config_jscsrc):
+def _get_all_files_in_directory(dir_path):
+    """Recursively collects all files in directory and 
+    subdirectories of specified path.
+
+    Args:
+    - dir_path: str. Path to the folder to be linted.
+
+    Returns:
+        a list of files in directory and subdirectories.
+    """
+    for subdirectory in os.listdir(dir_path):
+        subdirectory = os.path.join(dir_path,subdirectory)
+        if os.path.isdir(subdirectory):
+           _get_all_files_in_directory(subdirectory)
+        else:
+            files_in_directory.append(subdirectory)
+    return files_in_directory
+
+
+def _lint_js_files(node_path, jscs_path, config_jscsrc, input_dir):
     """Prints a list of lint errors in changed JavaScript files.
 
     Args:
     - node_path: str. Path to the node binary.
     - jscs_path: str. Path to the JSCS binary.
-    - autofix: bool. Whether to automatically fix errors.
+    - input_dir: str. Path to the folder to be linted (optional).
     """
-    changed_filenames = _get_changed_filenames()
-    changed_js_filenames = filter(_is_javascript_file, changed_filenames)
-    num_js_files = len(changed_js_filenames)
 
     num_files_with_errors = 0
 
     print '----------------------------------------'
 
-    if not changed_js_filenames:
-        print 'There are no JavaScript files to lint in this commit. Exiting.'
-        print '----------------------------------------'
-        sys.exit(0)
+    if input_dir:
+        if os.path.isfile(input_dir):
+            all_files = [input_dir]
+        else:
+            all_files = _get_all_files_in_directory(input_dir)
+
+        files_to_lint  = filter(_is_javascript_file, all_files)
+
+        num_js_files = len(files_to_lint)
```

Comment:
You can pull num_js_files to the bottom (after start_time = time.time()) since it's common to both branches.